src/EditorFeatures/CSharp/BlockCommentEditing/BlockCommentEditingCommandHandler.cs

```diff
@@ -0,0 +1,177 @@
+
+            var firstNonWhitespacePosition = currentLine.GetFirstNonWhitespacePosition() ?? -1;
+            if (firstNonWhitespacePosition == -1)
+            {
+                return null;
+            }
+
+            var currentLineStartsWithBlockCommentStartString = currentLine.StartsWith(firstNonWhitespacePosition, "/*", ignoreCase: false);
+            var currentLineStartsWithBlockCommentEndString = currentLine.StartsWith(firstNonWhitespacePosition, "*/", ignoreCase: false);
+            var currentLineStartsWithBlockCommentMiddleString = currentLine.StartsWith(firstNonWhitespacePosition, "*", ignoreCase: false);
+
+            if (!currentLineStartsWithBlockCommentStartString && !currentLineStartsWithBlockCommentMiddleString)
+            {
+                return null;
+            }
+
+            if (!IsCaretInsideBlockCommentSyntax(caretPosition))
+            {
+                return null;
+            }
+
+            if (currentLineStartsWithBlockCommentStartString)
+            {
+                if (BlockCommentEndsRightAfterCaret(caretPosition))
+                {
+                    //      /*|*/
+                    return " ";
+                }
+                else if (caretPosition == firstNonWhitespacePosition + 1)
+                {
+                    //      /|*
+                    return null; // The newline inserted could break the syntax in a way that this handler cannot fix, let's leave it.
+                }
+                else
+                {
+                    //      /*|
+                    return " *" + GetPaddingOrIndentation(currentLine, caretPosition, firstNonWhitespacePosition, "/*");
+                }
+            }
+
+            if (currentLineStartsWithBlockCommentEndString)
+            {
+                if (BlockCommentEndsRightAfterCaret(caretPosition))
+                {
+                    //      /*
+                    //      |*/
+                    return " ";
+                }
+                else if (caretPosition == firstNonWhitespacePosition + 1)
+                {
+                    //      *|/
+                    return "*";
+                }
+                else
+                {
+                    //      /*
+                    //   |   */
+                    return " * ";
+                }
+            }
+
+            if (currentLineStartsWithBlockCommentMiddleString)
+            {
+                if (BlockCommentEndsRightAfterCaret(caretPosition))
+                {
+                    //      *|*/
+                    return "";
+                }
+                else if (caretPosition > firstNonWhitespacePosition)
+                {
+                    //      *|
+                    return "*" + GetPaddingOrIndentation(currentLine, caretPosition, firstNonWhitespacePosition, "*");
+                }
+                else
+                {
+                    //      /*
+                    //   |   *
+                    return " * ";
```

Comment:
`GetPaddingOrIndentation` was decided to assume the caret position was behind the  `*`.